### ISSUE TITLE
kodi blanks display after a few minutes idle

### ISSUE THREAD

**Issue by spiderbit** [NixOS/nixpkgs#61719]
## Issue description
Just updated from 17.09 to 19.03 and now after no action for a while the screen goes dark, as a side issue I can't "wake it up" with my xbox controller, but with a keyboard I can.

But the real question is why does it fall asleep at all. I find nothing in kodi settings and I even added logind ignore behaviour:

``` 
services.logind.extraConfig = "IdleAction=ignore";
```
logind.conf 
```
[Login]
KillUserProcesses=no
HandleLidSwitch=suspend
HandleLidSwitchDocked=ignore
HandleLidSwitchExternalPower=suspend
IdleAction=ignore
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.42, NixOS, 19.03.172604.7cd2e4ebe8c (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.03.172604.7cd2e4ebe8c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by aanderse** [comment]
Does your xbox controller work with kodi at all? Which desktop environment are you running? We might need your configuration.nix to better troubleshoot.

**Comment by spiderbit** [comment]
Funny enough it does not only work (controller) but I could even kick out all xboxdrv code, which I tested after I wrote this bug report so the removel did not cause that which was that.

The desktop environment is xserver or do you mean the desktop manager? that would be kodi that should be the relevant parts of the config:
```
  environment.systemPackages = with pkgs; [
      nmap python3 youtube-dl libvdpau-va-gl vaapiVdpau glxinfo
      vdpauinfo certbot openssl retroarch kodiPlugins.advanced-launcher
      kodiPlugins.pdfreader zsnes dmidecode nixops
  ];

  # Enable the X11 windowing system.
  services.xserver = {
    enable = true;
    layout = "dvorak";
    xkbOptions = "eurosign:e";
    displayManager.lightdm.autoLogin = {
      user = "my-user";
      enable = true;
    };
    desktopManager.default = "kodi";
    desktopManager.kodi.enable = true;
  };  

  services.logind.extraConfig = "IdleAction=ignore";
  nixpkgs.config.KodiPlugin = {
    advanced-launcher = true;
    enablePeripheralJoystick = true;
    # enableControllers = true; # might not be needed  
  };

  nixpkgs.config = {
    kodi = {
      enableJoystick = true;
    };
  };

  hardware.opengl.enable = true;
  hardware.opengl.extraPackages = [ pkgs.libvdpau-va-gl pkgs.vaapiVdpau pkgs.vaapiIntel];
 
```

**Comment by spiderbit** [comment]
I just switched to xfce and it also goes to sleep after 10 mins or so. os it's not a kodi problem, except that the controller don't wakes the display, but that it blanks the screen is the major problem.

Btw I see a cursor (a x) at the end of a video, when I don't use a mouse which also makes not much sense.

**Comment by aanderse** [comment]
With regards to your cursor issue I have added the `unclutter-xfixes` package to my dedicated htpc setups and then execute it on login. This hides the cursor when the mouse isn't being moved.

As far as blank display you might consider trying something like [this](https://unix.stackexchange.com/a/38179).

**Comment by spiderbit** [comment]
That works as workaround:
```
    services.xserver.displayManager.setupCommands = ''
      /run/current-system/sw/bin/xset -dpms
      /run/current-system/sw/bin/xset s off
    '';
```
But I don't see that as a fix, while I could see a 10 min blank as a default that some people want, that the controller can't wake up the blankscreen is surely not a good default.

But I even wonder if that is a good default for kodi, it might make sense for xfce and gnome but for kodi?

**Comment by aanderse** [comment]
@spiderbit I definitely agree if the controller can't wake up the screen that is a problem. The controller works with kodi when the screen is on, right?

**Comment by spiderbit** [comment]
Yes

**Comment by aanderse** [comment]
@spiderbit If you find value in the above behaviour you could create a PR modifying https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/x11/desktop-managers/kodi.nix to add your snippet above.

As far as the controller goes... I assume you don't know if this was an issue with 17.09 because you didn't have the screen off issue?

Might we consider this issue resolved?

**Comment by spiderbit** [comment]
Well technically we could close this bug I guess but wouldn't you keep it open till such a PR was accepted?

But then we need a new bug for the not registered (to wake up) controller behavior, right?

How would you do include that code? disable power management completely in kodi generally? or give a boolean toggle for it? If you disable it hard coded then I guess this bug could be closed. I just wonder if some people could prefer the current behavior with the blanking?

Well I don't need it and till somebody finds a fix for the controller thing and wants that behavior it's probably better to deactivate the blanking (at least as default)

**Comment by aanderse** [comment]
Since you don't know if the controller would wake-up your screen in 17.09 I'm not entirely sure if this is a bug or just the way `xboxdrv` operates. I guess we should determine that first. Good point.

**Comment by spiderbit** [comment]
No, I don't use xboxdrv anymore at this moment, because apparently the normal evdev bits are enough to recognize my controller.

But the sleep bug was there before I changed that.

**Comment by cpages** [comment]
So I had the same issue using kore (the remote android app), and I could have sworn waking the screen worked in the past, but then I saw this post: https://forum.kodi.tv/showthread.php?tid=342934.

Long story short, I set xfce to never blank/shut the screen in the power saving options, and enabled it in kore (in power saving options as well), and now I can wake the screen.

I tested it also with my xbox controller, to ensure it was the same issue. Funny thing is moving up/down does not unlock the screen, but pressing a button does, so I hope this does the trick for you.
